Spigot-API-Patches/0269-Add-dropLeash-variable-to-EntityUnleashEvent.patch

```diff
@@ -0,0 +1,68 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: KennyTV <jahnke.nassim@gmail.com>
+Date: Fri, 29 Jan 2021 15:13:04 +0100
+Subject: [PATCH] Add dropLeash variable to EntityUnleashEvent
+
+
+diff --git a/src/main/java/org/bukkit/event/entity/EntityUnleashEvent.java b/src/main/java/org/bukkit/event/entity/EntityUnleashEvent.java
+index a33986a0c437a673435206fc337031a7eebdab3b..47e9b9e1ceb876e3ad701c4193dc02e7b30d27c3 100644
+--- a/src/main/java/org/bukkit/event/entity/EntityUnleashEvent.java
++++ b/src/main/java/org/bukkit/event/entity/EntityUnleashEvent.java
+@@ -10,10 +10,12 @@ import org.jetbrains.annotations.NotNull;
+ public class EntityUnleashEvent extends EntityEvent {
+     private static final HandlerList handlers = new HandlerList();
+     private final UnleashReason reason;
++    private boolean dropLeash; // Paper - drop leash variable
+ 
+-    public EntityUnleashEvent(@NotNull Entity entity, @NotNull UnleashReason reason) {
++    public EntityUnleashEvent(@NotNull Entity entity, @NotNull UnleashReason reason, boolean dropLeash) {  // Paper - drop leash variable
+         super(entity);
+         this.reason = reason;
++        this.dropLeash = dropLeash; // Paper - drop leash variable
+     }
+ 
+     /**
+@@ -26,6 +28,26 @@ public class EntityUnleashEvent extends EntityEvent {
+         return reason;
+     }
+ 
++    // Paper start
++    /**
++     * Returns whether a leash item will be dropped.
++     *
++     * @return Whether the leash item will be dropped
++     */
++    public boolean isDropLeash() {
++        return dropLeash;
++    }
++
++    /**
++     * Sets whether a leash item should be dropped.
++     *
++     * @return Whether the leash item should be dropped
++     */
++    public void setDropLeash(boolean dropLeash) {
++        this.dropLeash = dropLeash;
++    }
++    // Paper end
++
+     @NotNull
+     @Override
+     public HandlerList getHandlers() {
+diff --git a/src/main/java/org/bukkit/event/player/PlayerUnleashEntityEvent.java b/src/main/java/org/bukkit/event/player/PlayerUnleashEntityEvent.java
+index cf78950b56d4977f6c4d9d98d183bfc5ba3bacc0..305ad62fa9fba3313795603ddd00ba4466d699fc 100644
+--- a/src/main/java/org/bukkit/event/player/PlayerUnleashEntityEvent.java
++++ b/src/main/java/org/bukkit/event/player/PlayerUnleashEntityEvent.java
+@@ -13,8 +13,10 @@ public class PlayerUnleashEntityEvent extends EntityUnleashEvent implements Canc
+     private final Player player;
+     private boolean cancelled = false;
+ 
+-    public PlayerUnleashEntityEvent(@NotNull Entity entity, @NotNull Player player) {
+-        super(entity, UnleashReason.PLAYER_UNLEASH);
++    // Paper start - drop leash variable
++    public PlayerUnleashEntityEvent(@NotNull Entity entity, @NotNull Player player, boolean dropLeash) {
```

Comment:
API breaking change without any prior deprecation. Just like with the other constructor, keeping the old one and making it delegate to the new one would fix the compatibility issue.